src/util/invariant.cpp

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************\
+
+Module:
+
+Author: Martin Brain, martin.brain@diffblue.com
+
+\*******************************************************************/
+
+
+#include "invariant.h"
+
+#include <stdexcept>
+#include <string>
+#include <sstream>
+
+
+#ifdef CPROVER_INVARIANT_PRINT_STACK_TRACE
+#include <iostream>
+#include <stdlib.h>
+#endif
+
+// Backtraces compiler and C library specific
+#include <assert.h>
```

Comment:
Ah! Would you mind adding this as a comment to the code so that no one ever changes this in a naive cleanup attempt? Thanks!